flink-runtime/src/main/java/org/apache/flink/runtime/state/DefaultOperatorStateBackend.java

```diff
@@ -325,8 +325,7 @@ public OperatorStateHandle performOperation() throws Exception {
 		return task;
 	}
 
-	@Override
-	public void restore(Collection<OperatorStateHandle> restoreSnapshots) throws Exception {
+	public void restore(StateObjectCollection<OperatorStateHandle> restoreSnapshots) throws Exception {
```

Comment:
Why can't it remain a `Collection`?

Reply:
Because there where a couple of static helper methods operating on those collections, e.g. disposing a collection of state handles or checking if the collection as a whole has any state. IMO those things should go to a separate class and not be static helpers over a generic collection.

Reply:
Ok, if you just talk about this particular usage, I might double-check if the interface is enough.

Reply:
I couldn't find any call sites of these methods in the implementations of `Snapshotable#restore`. But I only skimmed over it.

Reply:
True, it can remain `Collection`.